training/src/app/chuck-noris/vip/vip.component.ts

```diff
@@ -0,0 +1,20 @@
+import { Component, OnInit } from '@angular/core';
+import { Joke } from "../joke";
+import { ActivatedRoute } from "@angular/router";
+
+@Component({
+	selector: 'app-vip',
+	templateUrl: './vip.component.html',
+	styleUrls: ['./vip.component.scss']
+})
+export class VipComponent implements OnInit {
+	public joke: Joke;
+
+	constructor(private activatedRoute: ActivatedRoute) {
+	}
+
+	public ngOnInit() {
+		console.log(this.activatedRoute.snapshot.data);
+		this.joke = this.activatedRoute.snapshot.data[ '0' ];
```

Comment:
'0' is a problem since the order might change some day.
u can give it a specific key inside the app routing, see my comment there, then fetch it here with that key

Reply:
also, remove console log when pushing code